### PR TITLE
Fix JSON bodies in state/http.

### PR DESCRIPTION
**Pull request by sgomes**
Objects weren't being correctly encoded into JSON.

This was causing issues with 2FA in the WordPress mobile app (see p4a5px-2oW-p2)

#### Changes proposed in this Pull Request

* Correctly encode object bodies into JSON.

#### Testing instructions

* [To be added by @koke]

### CODE REVIEW

**Comment by sgomes** [comment]
Note: it's still unclear whether this is the right approach, or whether it should be fixed at the consumer's level. @Tug is looking into this.

**Comment by sgomes** [comment]
One potential issue with this approach is that it fails to take into account some types that `fetch` can handle natively and shouldn't be serialised into JSON.

**Review by designsimply — COMMENTED** [review]
👍 I tested this using the live branch at https://calypso.live/?branch=fix/state-http-json-bodies and it worked in my test! ([36s](https://cloudup.com/cUMGCFIDx_2))

**Comment by koke** [comment]
Tested locally and it's working for me as well 👌

**Comment by sgomes** [comment]
@Tug Do you feel this is ready to go?

**Comment by Tug** [comment]
@sgomes Yes sir :shipit:

**Comment by sgomes** [comment]
Oops, still need an approval! 😅 

So you mind approving so I can submit, @Tug?

**Review by Tug — APPROVED** [review]
Feels weird accepting my own changes but well
👍

**Comment by sgomes** [comment]
Thanks, @Tug!

**Pull request merged**
